tests/jungle/corruption_test.cc

```diff
@@ -922,14 +922,182 @@ int duplicate_seq_flush_test() {
 #include "db_internal.h"
 #include "table_manifest.h"
 #include "table_mgr.h"
-
 #include <libjungle/jungle.h>
 
 // To access the internal structure of DB.
 namespace jungle {
 namespace checker {
 class Checker {
 public:
+
+static int compaction_empty_table_test() {
+    std::string filename;
+    TEST_SUITE_PREPARE_PATH(filename);
+
+    jungle::DB *db;
+    jungle::Status s;
+
+    // Open DB.
+    jungle::DBConfig config;
+    TEST_CUSTOM_DB_CONFIG(config);
+    config.purgeDeletedDocImmediately = false;
+    config.fastIndexScan = true;
+    config.minFileSizeToCompact = 1024;
+    config.bloomFilterBitsPerUnit = 10;
+    CHK_Z(jungle::DB::open(&db, filename, config));
+
+    //Inserting NUM_ENTRIES no of entries
+    const size_t NUM_DIGITS = 6;
+    const size_t NUM_ENTRIES = 10000;
+    for (size_t ii = 0; ii < NUM_ENTRIES; ++ii)
+    {
+        jungle::Record rec;
+        std::string key_str = "key" + TestSuite::lzStr(NUM_DIGITS, ii);
+        std::string meta_str = "meta" + TestSuite::lzStr(NUM_DIGITS, ii);
+        std::string value_str = "value" + TestSuite::lzStr(NUM_DIGITS, ii);
+        rec.kv.key = jungle::SizedBuf(key_str);
+        rec.kv.value = jungle::SizedBuf(value_str);
+        rec.meta = jungle::SizedBuf(meta_str);
+        CHK_Z(db->setRecordByKey(rec));
+    }
+
+    CHK_Z(db->sync(false));
+    CHK_Z(db->flushLogs());
+    for (size_t ii = 0; ii < 4; ++ii)
+    {
+        CHK_Z(db->compactL0(jungle::CompactOptions(), ii));
+    }
+
+
+    //Getting the information on the last table in level 1
+    std::list<jungle::TableInfo *> tables_out;
+    db->p->tableMgr->mani->getTablesRange(1, jungle::SizedBuf(), jungle::SizedBuf(), tables_out);
+    size_t table_size_before_deletion = tables_out.size();
+    jungle::TableInfo *t_last = nullptr;
+    auto itr = tables_out.begin();
+    for (size_t ii = 0; ii < tables_out.size(); ++ii)
+    {
+
+        if (ii == tables_out.size() - 1)
+        {
+            t_last = *itr;
+        }
+        itr++;
+    }
+
+    std::cout << "No of tables: " << table_size_before_deletion << "\n Last table minKey " << t_last->minKey.toReadableString() << "\n";
```

Comment:
Please keep a line shorter than 90 chars.
You can use `git-clang-format` with the `clang-format` file in our internal KV store repo.

Reply:
+) Instead of `std::cout`, you can use `TestSuite::Msg`:
```c++
TestSuite::Msg test_msg;
test_msg << "...";
```
then `test_msg` will be printed out only when `printTestMessage` is `true`.